File: Advanced Programming Methods/Toy Language C#/A2/Model/Expressions/VariableExpression.cs
using System;
using System.Collections.Generic;
using A2.Model.Data_Structures;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace A2.Model.Expressions
{
    //returns the value associated to id from the Symbol Table given as parameter
    //error if id not found
    public class VariableExpression : Expression{
        private string id;

        public VariableExpression(String id)
        {
            this.id = id;
        }
        public override int eval(Data_Structures.IDictionary<string, int> dict)
        {
            if(!dict.containsKey(id))
                throw new MyException("Variable not in symbol table!");
            return dict.get(id);
        }
            public override string ToString()
        {
            return id;
        }
    }
}

File: Advanced Programming Methods/Toy Language JavaFX/src/Model/Expressions/VariableExpression.java
package Model.Expressions;

import Model.DataStructures.IDictionary;
import Model.DataStructures.IHeap;
import Model.DataStructures.MyException;

//returns the value associated to id from the Symbol Table given as parameter
//error if id not found
public class VariableExpression extends Expression {
    private String id;

    public VariableExpression(String id){
        this.id = id;
    }

    @Override
    public int eval(IDictionary<String, Integer> dict, IHeap<Integer, Integer> heap) throws MyException {
        if(!dict.containsKey(id))
            throw new MyException("Variable not in symbol table!");
        return dict.get(id);
    }

    @Override
    public String toString() {
        return id;
    }
}
